element/index.js

```diff
@@ -209,3 +210,34 @@ export function RawHTML( { children, ...props } ) {
 		...props,
 	} );
 }
+
+/**
+ * Given a component returns the enhanced component augmented with a component
+ * only rerendering when its props/state change
+ *
+ * @param {Function} mapComponentToEnhancedComponent Function mapping component
+ *                                                   to enhanced component.
+ * @param {string}   modifierName                    Seed name from which to
+ *                                                   generated display name.
+ *
+ * @return {WPComponent} Component class with generated display name assigned.
+ */
+export const purify = createHigherOrderComponent( ( Wrapped ) => {
```

Comment:
Why did you choose to export this in `element`, and not `components` ?

FWIW, I had a branch with similar change, and encountered that `data` would then depend on `components`, which surfaced yet another issue of: Should `withSelect` and `withDispatch` be in `data` or not?

Reply:
I asked myself several times about `withFilters` in `components` which depends on `hooks`. A very similar case.

> Why did you choose to export this in element, and not components ?

I'm sure this is all about dependencies. There needs to be a solution :)

Reply:
I actually didn't think that much about it. I put it under element to match `React` :) which also provides a pure component definition.

Reply:
Well explained, I'm sold :)

Reply:
That's definitely a good question and I think having it in `element` is probably not ideal. It seems that we struggle a lot with the `components` module but it contains too much stuff. I wonder for instance if we should try to split it into smaller modules:

 - For example `pure/ifCondition/withState` into something called `compose` :P (aka recompose)
 - `withFocusReturn/NavigableContainer (without UI)` into `a11y`
 - 'Button/IconButton/Popover/MenuItems/DropDown' into `ui`

That's not a well-formed thinking, just some quick ideas

Reply:
`pure`, `ifCondition`, `withState`, `withInstanceId`, but also `createHigherOrderComponent` and `compose` - we can put all of them together in their own module 💯

Reply:
So did we just not decide anything here?

Reply:
I assumed it's going to be renamed to `pure`.

Reply:
It was ... I guess you are referring to the location of this HOC, we need to come up with something sooner than later 👍